boards/arm/nucleo_wb55cg_dongle/doc/nucleo_wb55cg_dongle.rst

```diff
@@ -0,0 +1,296 @@
++-----------+------------+-------------------------------------+
+| PWM       | on-chip    | pwm                                 |
++-----------+------------+-------------------------------------+
+| ADC       | on-chip    | adc                                 |
++-----------+------------+-------------------------------------+
+| WATCHDOG  | on-chip    | independent watchdog                |
++-----------+------------+-------------------------------------+
+| RADIO     | on-chip    | Bluetooth Low Energy                |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on this Zephyr port.
+
+The default configuration can be found in the defconfig file:
+``boards/arm/nucleo_wb55cg_dongle/nucleo_wb55cg_dongle_defconfig``
+
+Bluetooth and compatibility with STM32WB Copro Wireless Binaries
+================================================================
+
+To operate bluetooth on Nucleo WB55RG, Cortex-M0 core should be flashed with
+a valid STM32WB Coprocessor binaries (either 'Full stack' or 'HCI Layer').
+These binaries are delivered in STM32WB Cube packages, under
+Projects/STM32WB_Copro_Wireless_Binaries/STM32WB5x/
+To date, interoperability and backward compatibility has been tested and is
+guaranteed up to version 1.5 of STM32Cube package releases.
+
+Connections and IOs
+===================
+
+Nucleo WB55RG Board has 6 GPIO controllers. These controllers are responsible for pin muxing,
+input/output, pull-up, etc.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+.. rst-class:: rst-columns
+
+- UART_1 TX/RX : PB7/PB6
+- LPUART_1 TX/RX : PA3/PA2 (arduino_serial)
+- I2C_1_SCL : PB8
+- I2C_1_SDA : PB9
+- I2C_3_SCL : PC0
+- I2C_3_SDA : PC1
+- USER_PB : PC4
+- USER_PB1 : PD0
+- USER_PB2 : PD1
+- LD1 : PB5
+- LD2 : PB0
+- LD3 : PB1
+- SPI_1_NSS : PA4 (arduino_spi)
+- SPI_1_SCK : PA5 (arduino_spi)
+- SPI_1_MISO : PA6 (arduino_spi)
+- SPI_1_MOSI : PA7 (arduino_spi)
+- PWM_2 CH 1 : PA0
+- ADC_1_CH3 : PC2
+
+System Clock
+------------
+
+Nucleo WB55RG System Clock could be driven by internal or external oscillator,
+as well as main PLL clock. By default System clock is driven by HSE clock at 32MHz.
+
+Serial Port
+-----------
+
+Nucleo WB55RG board has 2 (LP)U(S)ARTs. The Zephyr console output is assigned to USART1.
+Default settings are 115200 8N1.
+
+
+Programming and Debugging
+*************************
+
+Applications for the ``nucleo_wb55cg_dongle`` board configuration can be built the
+usual way (see :ref:`build_an_application`).
+
+Flashing
+========
+
+Nucleo WB55RG board includes an ST-LINK/V2-1 embedded debug tool
+interface.  This interface is supported by the openocd version included in the
+Zephyr SDK since v0.11.0.
+
+If you prefer, you can use pyocd, but it requires to enable "pack" support with
+the following pyocd command:
+
+.. code-block:: console
+
+   $ pyocd pack --install stm32wb55cg
+
```

Comment:
```suggestion
The Nucleo WB55 USB Dongle does not include an embedded debug tool interface.
However it supports the following programming options:

1. Using the ROM bootloader
2. Using MCUboot in serial recovery mode
3. Using an external :ref:`debug probe <debug-probes>`

The different options can be used with the `west <west>` tool
and assume you are in the root directory of your :term:`west installation`.

Option 1: Using the ROM bootloader and dfu-util
----------------------------------
Flashing using the ROM bootloader requires a special activation pattern,
which can be triggered by pulling the BOOT0 pin high to 3.3V.
In case of the Nucleo WB55 USB Dongle this can be achieved by switching `BOOT0`
into 1 position(away from the USB port).
The ROM bootloader supports flashing via the USB port(DFU), and additionally via
UART, I2C and SPI.
You can read more about how to enable and use the ROM bootloader by checking the application note AN2606.

#. Installing dfu-util

It is recommended to use at least v0.8 of dfu-util.
The package available in Debian and Ubuntu can be quite old, so you might have
to build dfu-util from source. Information about how to get the source code
and how to build it can be found at the `DFU-UTIL website`_

#. Flash using dfu-util

Make sure BOOT0 is pulled high, then you can flash using west.
Here is an example for the :ref:`blinky-sample` application.

.. zephyr-app-commands::
   :zephyr-app: samples/basic/blinky
   :board: nucleo_wb55_usbdongle
   :goals: build flash



```

Reply:
You are mentioning pyocd, although it is not enabled in board.cmake.
As no debug probe is on the dongle, I would be more explicit about the flashing options here.
Describing the process of dfu-util, and eventually a reference how one could use mcuboot for easier flashing.